### PR TITLE
Use fixed length millisecond timestamp format for logs

### PR DESCRIPTION
**Pull request by sylr**
Port https://github.com/prometheus/prometheus/pull/5225 over to common

### CODE REVIEW

File: promlog/log.go [hunk 1]
```diff
@@ -18,12 +18,23 @@ package promlog
...
 	"github.com/go-kit/kit/log"
 	"github.com/go-kit/kit/log/level"
 	"github.com/pkg/errors"
 )
 
+var (
+	timestampFormat = "2006-01-02T15:04:05.000Z07:00"
+)
+
+func init() {
```
Comment (brian-brazil) [inline promlog/log.go]: You're still affecting all users of the library here. Can you make it so it only affects our loggers?

Comment (sylr) [reply promlog/log.go]: I don't think so, the global var is the only way to specify the timestamp format in the go-kit/log lib.

Comment (brian-brazil) [reply promlog/log.go]: You can change it 3 lines from the bottom of this file

Comment (sylr) [reply promlog/log.go]: Indeed

File: promlog/log.go [hunk 2]
```diff
@@ -18,12 +18,20 @@ package promlog
...
 
 	"github.com/go-kit/kit/log"
 	"github.com/go-kit/kit/log/level"
 	"github.com/pkg/errors"
 )
 
+var (
+	timestampFormat = log.TimestampFormat(
+		func() time.Time { return time.Now().UTC() },
+		"2006-01-02T15:04:05.000Z07:00",
```
Comment (brian-brazil) [inline promlog/log.go]: You should probably add a comment as to how this is different to the usual

Comment (sylr) [reply promlog/log.go]: Done

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**